### ISSUE TITLE
Bug: #995 not resolved, see details there

### ISSUE THREAD

**Issue by mtressl** [OpenWonderLabs/homebridge-switchbot#1000]
### Describe the Bug

Bug: #995 not resolved, see details there

@donavanbecker: Is it necessary to remove cached devices?

### To Reproduce

Bug: #995 not resolved, see details there

### Expected Behavior

Bug: #995 not resolved, see details there

### Relevant Log Output

```shell
[7/7/2024, 9:48:52 AM] [homebridge-switchbot] Restoring existing accessory from cache: Büro Synology deviceId: CF911xxx6559
[7/7/2024, 9:48:52 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Fan Service
[7/7/2024, 9:48:52 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Lock Service
[7/7/2024, 9:48:52 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Door Service
[7/7/2024, 9:48:52 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Faucet Service
[7/7/2024, 9:48:52 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Outlet Service
[7/7/2024, 9:48:52 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Window Service
[7/7/2024, 9:48:52 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Garage Door Service
[7/7/2024, 9:48:52 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Window Covering Service
[7/7/2024, 9:48:52 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Stateful Programmable Switch Service
[7/7/2024, 9:48:52 AM] [homebridge-switchbot] Restoring existing accessory from cache: Büro Vorhang deviceId: C985Exxx3C39
[7/7/2024, 9:48:52 AM] [homebridge-switchbot] Restoring existing accessory from cache: Schlafzimmer Vorhang deviceId: CAD91xxxBC05
[7/7/2024, 9:48:54 AM] [homebridge-switchbot] Curtain: Schlafzimmer Vorhang failed openAPIRefreshStatus with OpenAPI Connection, Error Message: "Cannot use 'in' operator to search for 'power' in undefined"
[7/7/2024, 9:48:54 AM] [homebridge-switchbot] Curtain: Büro Vorhang failed openAPIRefreshStatus with OpenAPI Connection, Error Message: "Cannot use 'in' operator to search for 'power' in undefined"
```

### Config for homebridge-switchbot

```shell
Bug: #995 not resolved, see details there
```

### Device and Model

Bug: #995 not resolved, see details there

### Node.js Version

Bug: #995 not resolved, see details there

### NPM Version

Bug: #995 not resolved, see details there

### Homebridge Version

Bug: #995 not resolved, see details there

### Homebridge Switchbot Plugin Version

Bug: #995 not resolved, see details there

### Operating System

Bug: #995 not resolved, see details there

**Comment by donavanbecker** [comment]
Can you try the latest beta: 

[![npm version](https://badgen.net/badge/homebridge-switchbot@beta/>=3.6.1-beta.2/orange)](https://www.npmjs.com/package/@switchbot/homebridge-switchbot)

[Click here to find out how to install Beta](https://github.com/OpenWonderLabs/homebridge-switchbot/wiki/Beta-Version)

**Comment by mtressl** [comment]
hi @donavanbecker 
missed the beta.2, but tried beta.3 instead. Error still coming up after the restart, this is the log:

`[7/13/2024, 11:53:42 AM] Registering platform '@switchbot/homebridge-switchbot.SwitchBot'
[7/13/2024, 11:53:42 AM] [homebridge-switchbot] Loaded @switchbot/homebridge-switchbot **v3.6.1-beta.3** child bridge successfully
[7/13/2024, 11:53:42 AM] Loaded 3 cached accessories from cachedAccessories.0E8E8xxx54EE.
[7/13/2024, 11:53:42 AM] Homebridge v1.8.3 (HAP v0.12.2) (homebridge-switchbot) is running on port 43358.
[7/13/2024, 11:53:43 AM] [homebridge-switchbot] Total SwitchBot Devices Found: 4
[7/13/2024, 11:53:43 AM] [homebridge-switchbot] Restoring existing accessory from cache: Büro Synology deviceId: CF9118456559
[7/13/2024, 11:53:43 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Fan Service
[7/13/2024, 11:53:43 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Lock Service
[7/13/2024, 11:53:43 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Door Service
[7/13/2024, 11:53:43 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Faucet Service
[7/13/2024, 11:53:43 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Outlet Service
[7/13/2024, 11:53:43 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Window Service
[7/13/2024, 11:53:43 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Garage Door Service
[7/13/2024, 11:53:43 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Window Covering Service
[7/13/2024, 11:53:43 AM] [homebridge-switchbot] Bot: Büro Synology Removing any leftover Stateful Programmable Switch Service
[7/13/2024, 11:53:43 AM] [homebridge-switchbot] Restoring existing accessory from cache: Büro Vorhang deviceId: C985xxxB3C39
[7/13/2024, 11:53:43 AM] [homebridge-switchbot] Restoring existing accessory from cache: Schlafzimmer Vorhang deviceId: CAD9xxx5BC05
**[7/13/2024, 11:53:44 AM] [homebridge-switchbot] Curtain: Büro Vorhang failed openAPIRefreshStatus with OpenAPI Connection, Error Message: "Cannot use 'in' operator to search for 'power' in undefined"
[7/13/2024, 11:53:44 AM] [homebridge-switchbot] Curtain: Schlafzimmer Vorhang failed openAPIRefreshStatus with OpenAPI Connection, Error Message: "Cannot use 'in' operator to search for 'power' in undefined"**`

**Comment by mtressl** [comment]
v3.6.1-beta.15 introduced a new WARNING, and does not fix the problem in #995 

`[7/14/2024, 10:10:02 AM] [homebridge-switchbot] WARNING: The accessory 'Büro Vorhang' has an invalid 'configDeviceName' characteristic ('Büro Vorhang'). Please use only alphanumeric, space, and apostrophe characters. Ensure it starts and ends with an alphabetic or numeric character, and avoid emojis. This may prevent the accessory from being added in the Home App or cause unresponsiveness.
[7/14/2024, 10:10:02 AM] [homebridge-switchbot] Removing invalid characters from 'configDeviceName' characteristic`

**Comment by mtressl** [comment]
*bump*

Not resolved in latest v3.7.0

**Comment by donavanbecker** [comment]
Can you try the latest beta: 

[![npm version](https://badgen.net/badge/homebridge-switchbot@beta/>=3.7.1-beta.6/orange)](https://www.npmjs.com/package/@switchbot/homebridge-switchbot)

[Click here to find out how to install Beta](https://github.com/OpenWonderLabs/homebridge-switchbot/wiki/Beta-Version)
